File: services/orders-service/src/main/java/pl/xsware/orders/application/event/PaymentSucceededEvent.java
package pl.xsware.orders.application.event;

import java.time.Instant;
import java.util.UUID;

public record PaymentSucceededEvent(
    UUID eventId,
    String eventType,
    int version,
    Instant occurredAt,
    Data data
) {

    public static final String TYPE = "PaymentSucceeded";
    public static final int VERSION = 1;

    public record Data(
        UUID orderId,
        UUID paymentId,
        String provider,
        String externalId
    ) {}
}

File: services/payments-service/src/main/kotlin/pl/xsware/payments/application/event/PaymentSucceededEvent.kt
package pl.xsware.payments.application.event

import java.math.BigDecimal
import java.time.Instant
import java.util.UUID

data class PaymentSucceededEvent(
    val eventId: UUID,
    val eventType: String,
    val version: Int,
    val occurredAt: Instant,
    val data: Data
) {
    companion object {
        const val TYPE = "PaymentSucceeded"
        const val VERSION = 1
    }

    data class Data(
        val paymentId: UUID,
        val orderId: UUID,
        val amount: BigDecimal,
        val currency: String,
        val provider: String,
        val externalId: String
    )
}
